File: 3773-minimum-pair-removal-to-sort-array-i/minimum-pair-removal-to-sort-array-i.cpp
class Solution {
public:
    int minimumPairRemoval(vector<int>& nums) {
        int n = nums.size();
        bool non_decreasing;
        int count = 0;
        do {
            non_decreasing = true;
            int min_i = n-1;
            for (int i = n-1; i > 0; i--) {
                cout << nums[i] << " " << non_decreasing << endl;
                if (nums[i-1] > nums[i]) non_decreasing = false;
                if (nums[i] + nums[i-1] <= nums[min_i] + nums[min_i - 1]) min_i = i;
            }
            if (non_decreasing) break;
            nums[min_i] = nums[min_i] + nums[min_i-1];
            nums.erase(nums.begin() + min_i-1);
            n--;
            count++;
        } while (!non_decreasing);
        return count;
    }
};

File: 3773-minimum-pair-removal-to-sort-array-i/minimum-pair-removal-to-sort-array-i.py
class Solution:
    def isIncreasing(self, nums):
        for i in range(len(nums) - 1):
            if nums[i] > nums[i+1]:
                return False;
        return True;

    def minimumPairRemoval(self, nums: List[int]) -> int:
        ops = 0;
        while (not self.isIncreasing(nums)):
            min_sum = nums[0] + nums[1];
            min_i = 0;
            for i in range(len(nums) - 1):
                pair_sum = nums[i] + nums[i+1];
                if pair_sum < min_sum:
                    min_i = i;
                    min_sum = pair_sum;
            nums.pop(min_i + 1);
            nums[min_i] = min_sum;
            ops += 1;
        return ops;

            